src/ProjectOrigin.PedersenCommitment/Ristretto/SigmaProof.cs

```diff
@@ -26,15 +26,18 @@ private ZeroProof(Scalar c, Scalar z)
     public static ZeroProof Prove(Generator gen, Scalar r, byte[] label)
     {
         var c0 = gen.H() * r;
-        var oracle = (Point A) => Oracle(label, A, c0, gen.G(), gen.H());
+        var oracle = new Oracle(label);
+        oracle.Add(c0);
         return ZeroProof.Prove(gen, r, oracle);
     }
 
-    internal static ZeroProof Prove(Generator gen, Scalar r, Func<Point, Scalar> oracle)
+    internal static ZeroProof Prove(Generator gen, Scalar r, Oracle oracle)
     {
         var a = Scalar.Random();
         var A = gen.H() * a;
-        var c = oracle(A);
+        oracle.Domain("ZeroProof");
```

Comment:
These domain separation strings at the beginning of each proof are a good idea.

Reply:
Absolutely - it makes it very explicit what we are doing as well